modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/PublisherCreateWebAppTestCase.java

```diff
@@ -0,0 +1,98 @@
+    private String trackingCode = "AM_PublisherCreateWebAppTestCase";
+    private String appVersion = "1.0.0";
+    private String backEndUrl;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void startUp() throws Exception {
+        AutomationContext appMServer = new AutomationContext(AppmTestConstants.APP_MANAGER,
+                                                             TestUserMode.SUPER_TENANT_ADMIN);
+        backEndUrl = appMServer.getContextUrls().getWebAppURLHttps();
+        appmPublisherRestClient = new APPMPublisherRestClient(backEndUrl);
+        User adminUser = appMServer.getSuperTenant().getTenantAdmin();
+        appmPublisherRestClient.login(adminUser.getUserName(), adminUser.getPassword());
+    }
+
+    @Test(description = TEST_DESCRIPTION)
+    public void testPublisherCreateWebApp() throws Exception {
+        //Policy Adding before app creating.
+        String appDescription = "default app description for " + appName;
+        HttpResponse policyAddingResponse = appmPublisherRestClient.addPoicyGroup(appDescription);
+        assertTrue(policyAddingResponse.getResponseCode() == 200, "Non 200 status code received.");
+        String httpResponseData = policyAddingResponse.getData();
+        JSONObject responseData = new JSONObject(httpResponseData);
+        String response = responseData.getString(AppmTestConstants.RESPONSE);
+        JSONObject responseObject = new JSONObject(response);
+        String policyId = responseObject.getString(AppmTestConstants.ID);
+        String policyGroupId = "[" + policyId + "]";
+
+        //Web app create.
+        AppCreateRequest appRequest = new AppCreateRequest(appName, context, appVersion,
+                                                           trackingCode);
+        appRequest.setUriTemplatePolicyGroupId0(policyId);
+        appRequest.setUriTemplatePolicyGroupId1(policyId);
+        appRequest.setUriTemplatePolicyGroupId2(policyId);
+        appRequest.setUriTemplatePolicyGroupId3(policyId);
+        appRequest.setUriTemplatePolicyGroupId4(policyId);
+        appRequest.setUriTemplatePolicyGroupIds(policyGroupId);
+        HttpResponse appCreateResponse = appmPublisherRestClient.createApp(appRequest);
+
+        assertTrue(appCreateResponse.getResponseCode() == 200, "Non 200 status code received.");
+        JSONObject appCreateResponseData = new JSONObject(appCreateResponse.getData());
+        assertEquals(appCreateResponseData.getString(AppmTestConstants.MESSAGE), "asset added",
+                                   "Asset has not added successfully");
+        assertNotNull(appCreateResponseData.getString(AppmTestConstants.ID), "app id is null");
+
+        // Create Service provider for the created web app.
+        HttpResponse ssoProviderAddedResponse = appmPublisherRestClient.addSsoProvider(appRequest);
+        assertTrue(ssoProviderAddedResponse.getResponseCode() == 200, "Non 200 status code received.");
```

Comment:
add http response codes to constant, print the retrieved response code in error message